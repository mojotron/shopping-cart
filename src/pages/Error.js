import React from "react";

const Error = () => {
  return (
    <section>
      <h1>Error page</h1>
    </section>
  );
};

export default Error;
