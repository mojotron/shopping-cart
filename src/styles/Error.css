.Error {
  height: 100vh;
  width: 100%;
  color: white;
  background-color: black;
  background-image: url("../images/wallpapersden.com_76859-1920x1080.jpg");
  background-size: cover;
  background-repeat: no-repeat;
}
.Error__heading {
  font-size: 3.5rem;
  padding: 10rem 0 0 5rem;
  text-shadow: 1px 1px 2px red, 0 0 1em rgb(255, 72, 0),
    0 0 0.2em rgb(255, 60, 0);
}
.Error__home-link {
  color: white;
  padding: 3rem 0 0 8rem;
  text-decoration: underline;
  font-size: 1.6rem;
  cursor: pointer;
}
.Error__home-link:hover {
  opacity: 0.5;
}
