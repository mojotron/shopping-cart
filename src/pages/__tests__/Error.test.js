import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Error from "../Error";

const MockError = () => {
  return (
    <BrowserRouter>
      <Error />
    </BrowserRouter>
  );
};

describe("Error page", () => {
  test("render Error snapshot", () => {
    const { container } = render(<MockError />);
    expect(container).toMatchSnapshot();
  });

  test("has heading with text 404 page not found", () => {
    render(<MockError />);
    expect(screen.getByRole("heading").textContent).toBe("404 Page not Found");
  });

  test("has link to home page", () => {
    render(<MockError />);
    expect(screen.getByText("back to home")).toBeInTheDocument();
  });
});
