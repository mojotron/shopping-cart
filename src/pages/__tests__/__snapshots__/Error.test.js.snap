// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page render Error snapshot 1`] = `
<div>
  <section
    class="Error"
  >
    <h1
      class="Error__heading"
    >
      404 Page not Found
    </h1>
    <a
      class="Error__home-link"
      href="/"
    >
      back to home
    </a>
  </section>
</div>
`;
